### PR TITLE
docs: initial upload of styles used by doc linter

### PR DESCRIPTION
**Pull request by bob-watson**
Adds styles and configuration files for contributors to use the Vale documentation linter in their IDE while they edit angular.io topics.
This commit includes documentation for using the documentation linter in the README.md file.
This commit does not alter any angular.io or other published documentation.

### CODE REVIEW

**Comment by bob-watson** [comment]
This commit is going to cause problems with the inclusive language CI test because it includes files with all the prohibited words in its tests to look for those words in the documentation. 

Is there a way to mark those file(s) as allowable exceptions to this CI test? It looks like it's only two files:
* aio/tools/doc-linter/dictionaries/en_us/en_US-web.dic
* aio/tools/doc-linter/styles/Angular/Angular_Inclusive.yml

**Review by josmar-crwdstffng — APPROVED** [review]
@LGTM
Reviewed-for: global-docs-approvers

I have confirmed that the definitions work as intended and match the standalone repo.

File: aio/tools/doc-linter/README.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+2. View the problems in your IDE and fix them.
+ 
+Log any problem you encounter with the documentation linter as [a new docs issue](https://github.com/angular/angular/issues/new?assignees=&labels=&template=3-docs-bug.yaml) in the repo. Give your new issue the `comp: docs` and `docsarea: linter` labels.
+
+## Fixing documentation problems identified by Vale
+
+Vale evaluates a markdown (`.md`) file when you open it in the IDE and each time you save it.
+
+Fixing the problems that Vale identifies can range from simple word substitutions to rewriting sentences or paragraphs.
+Keep in mind that the goal is to make your documentation  as clear and easy to read as you can.
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:26]:
```suggestion
Keep in mind that the goal is to make your documentation as clear and easy to read as you can.
```

File: aio/tools/doc-linter/README.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+Complete these procedures to install the Vale documentation linter on a MacOS system for the first time.
+To install Vale in another IDE or on another operating system, see [Vale documentation](https://vale.sh/docs/vale-cli/installation/).
+
+### Prerequisites
+These instructions assume your development computer is a MacOS system.
+
+**Before you proceed make sure that:**
+
+1. You have installed the [`angular/angular` repo](https://github.com/angular/angular).
+1. You have a development or writing environment for `angular.io` in the `angular/angular` repo and you can build the `angular.io` docs. For information about creating the authoring environment for `angular.io`, see the [Angular documentation project](https://github.com/bob-watson/angular/tree/doc-linter/aio#angular-documentation-project-httpsangulario)
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:131]: Inconsistent numbering. Most other numbered lists use increasing numbers.

File: aio/tools/doc-linter/README.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+2. You have [homebrew](https://brew.sh/) installed on your development computer.
+
+### Install Vale for the first time
+
+**On a Mac, locally run these tools:**
+
+1. Open Chrome.
+1. Open a command-line tool, such as `terminal`.
+1. Open VS Code.
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:140]: Inconsistent numbering. Most other numbered lists use increasing numbers.

```suggestion
1. Open Chrome.
2. Open a command-line tool, such as `terminal`.
3. Open VS Code.
```

File: aio/tools/doc-linter/README.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+Complete these procedures to install the Vale documentation linter on a MacOS system for the first time.
+To install Vale in another IDE or on another operating system, see [Vale documentation](https://vale.sh/docs/vale-cli/installation/).
+
+### Prerequisites
+These instructions assume your development computer is a MacOS system.
+
+**Before you proceed make sure that:**
+
+1. You have installed the [`angular/angular` repo](https://github.com/angular/angular).
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:130]: The term "installed" is a little ambiguous here. You can't really install a git repo.
Did you mean (a) [clone the `angular/angular` repo][1], (b) clone [your fork][2] of the `angular/angular` repo or (c) having cloned the repo (or fork) to run `yarn install` (to install dependencies)?

[1]: https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository
[2]: https://docs.github.com/en/get-started/quickstart/fork-a-repo
[3]:

File: aio/tools/doc-linter/README.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+<!-- vale Angular.WriteGood_Illusions = NO -->
+<!-- vale Angular.Angular_CliReference = NO -->
+
+**To install the VS Code Vale extension, in VS Code:**
+
+1. Install the `Vale` VS Code extension.
+     1. Search for `Vale`.
+     2. Install `Vale`.
+2. Navigate to **Preferences > Settings**. The **Preferences** menu item is located under the **Code** nav bar on macOS.
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:174]: Inconsistent casing of `macOS`. Most other occurrences in this document use `MacOS` (although I believe `macOS` is more correct).

File: aio/tools/doc-linter/README.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+<!-- vale Angular.Angular_CliReference = NO -->
+
+1. Run `brew install vale`.
+2. Ensure that Vale is available on your `$PATH` and allow you to stay up to date with new releases.
+   1. Get the path to the `vale` application to add as the value for **Vale > Vale CLI : Path**.
+   2. In the command-line tool, run `which vale`. Save the returned path for use in the next procedure.
+   3. If this command does not return a path, restart your Mac and repeat this step.
+3. Update the instance of the `angular` repo on your system.
+4. In the command line tool:
+   1. Navigate to your `github` workspace directory.
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:156]: Did you mean "git workspace" or "git working directory" (here and below)?
I don't think "github workspace directory" is a thing (but I might be wrong).

File: aio/tools/doc-linter/README.md [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+   2. In your `github` workspace, navigate to the `angular/aio/tools/doc-linter` directory.
+   3. In the `angular/aio/tools/doc-linter` directory, confirm that you have the `vale.ini` file.
+   4. Run `pwd` to get the path to the `vale.ini` file for use the next procedure.
```
Comment (gkalpak) [inline aio/tools/doc-linter/README.md:159]:
```suggestion
   4. Run `pwd` to get the path to the `vale.ini` file for use in the next procedure.
```

File: aio/tools/doc-linter/styles/Angular/Angular_Code-as-word.yml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# tests for strings qualified by back-ticks are surrounded by spaces or punctuation
```
[the remark this answers is not in the document]
Comment (bob-watson) [reply aio/tools/doc-linter/styles/Angular/Angular_Code-as-word.yml:1]: Thanks for catching this!

File: aio/tools/doc-linter/README.md [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+[Vale](https://vale.sh/) is an open source, command line tool that evaluates documentation for style and phrasing. It can help improve readability and the consistency of terminology and voice across a range of topics and contributors.
+
+The lint tool is helpful and has been tailored to the Angular documentation style guide. Unfortunately, it's not as powerful as the grammar and spelling features you find in contemporary document editors. It's just one more tool use along with the reviewers that you're already using to help make it easier to write great documentation.
+
+**To use Vale in your documentation:**
+
+1. [Install Vale](#install-vale-in-your-vs-code-ide) in your VS Code IDE or, if you use a different IDE, see the [Vale documentation](https://vale.sh/docs/vale-cli/installation/) for information about installing and adding it.
+2. View the problems in your IDE and fix them.
+ 
+Log any problem you encounter with the documentation linter as [a new docs issue](https://github.com/angular/angular/issues/new?assignees=&labels=&template=3-docs-bug.yaml) in the repo. Give your new issue the `comp: docs` and `docsarea: linter` labels.
```
[the remark this answers is not in the document]
Comment (bob-watson) [reply aio/tools/doc-linter/README.md:19]: Thanks. I rewrote this to be appropriate to a wider audience.

File: aio/tools/doc-linter/README.md (hunk 1 above)
Comment (bob-watson) [reply aio/tools/doc-linter/README.md:26]: Removed this in the current edit.

File: aio/tools/doc-linter/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (bob-watson) [reply aio/tools/doc-linter/README.md:131]: I did. Thanks!

Comment (bob-watson) [reply aio/tools/doc-linter/README.md:131]: updated

File: aio/tools/doc-linter/README.md (hunk 5 above)
Comment (bob-watson) [reply aio/tools/doc-linter/README.md:174]: fixed.

File: aio/tools/doc-linter/README.md (hunk 7 above)
Comment (bob-watson) [reply aio/tools/doc-linter/README.md:159]: fixed

File: aio/tools/doc-linter/vale.ini [hunk 10]
```diff
@@ -0,0 +1,17 @@
+StylesPath = ./styles
+MinAlertLevel = suggestion
+
+SkippedScopes = code-example, code-tabs, code-pane
+# IgnoredScopes = code-example, code-tabs, code-pane
+
+[*.md]
+
+BlockIgnores = (?m) *(<code-example[^>]*>[\s\S\n]*<\/code-example>)
```
[the remark this answers is not in the document]
Comment (bob-watson) [reply aio/tools/doc-linter/vale.ini:9]: See previous comment.

File: aio/tools/doc-linter/styles/Angular/Angular_Note.yml [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+extends: existence
+message: 'Alert heading must be bold.'
+link: https://angular.io/guide/docs-style-guide#alerts-and-callouts
+level: error
+ignorecase: false
+scope: raw
+raw:
+  - (?<!\*\*)NOTE[^\*]*(?!\*\*)
+  - (?<!\*\*)TIP[^\*]*(?!\*\*)
+  - (?<!\*\*)IMPORTANT[^\*]*(?!\*\*)
```
[the remark this answers is not in the document]
Comment (bob-watson) [reply aio/tools/doc-linter/styles/Angular/Angular_Note.yml:11]: I'd like to tighten them up under a separate issue/PR if that would work for you.
These currently work(-ish) for our content, but I agree that they could be improved.

The intention is to find cases of `NOTE|TIP|IMPORTANT` that are:
* all caps
* the only word on the line
* not wrapped by bold notation `**` on both sides

File: aio/tools/doc-linter/README.md (hunk 6 above)
Comment (bob-watson) [reply aio/tools/doc-linter/README.md:156]: It should be (and now is) *workspace directory*. Thanks!

**Comment by bob-watson** [comment]
The commits in this branch got all tangled up and look worse than the actual changes that were made.
Closing this PR and replacing with PR #46897.

**Pull request closed without merging**